netbox_inventory/models.py

```diff
@@ -238,6 +238,19 @@ def installed_device(self):
         else:
             return None
 
+    @property
+    def current_site(self):
+        location = self.current_location
+        if location:
+            return location.site
```

Comment:
If an asset is assigned to a device and that device is assigned to a site but not to a location, then this returns None.

I suggest changing the method to start from `self.installed_device` and then accessing device.site.

btw - currently current_site and installed_site always return the same result. I suggest you return `self.storage_location.site` if asset is not currently assigned. This way you always get a site the asset is supposed to be at.

Reply:
I've now applied the same logic as for the locations. This avoids accessing the device at multiple code locations: Both site and location are accessed first by the installed device, then by storage.